app/src/main/java/io/stormbird/wallet/service/KeyService.java

```diff
@@ -0,0 +1,1412 @@
+        try
+        {
+            File file = new File(path);
+            fin = new FileInputStream(file);
+            bytes = readBytesFromStream(fin);
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+        return bytes;
+    }
+
+
+    public synchronized static String getFilePath(Context context, String fileName)
+    {
+        return new File(context.getFilesDir(), fileName).getAbsolutePath();
+    }
+
+    private boolean writeBytesToFile(String path, byte[] data)
+    {
+        FileOutputStream fos = null;
+        try
+        {
+            File file = new File(path);
+            fos = new FileOutputStream(file);
+            // Writes bytes from the specified byte array to this file output stream
+            fos.write(data);
+            return true;
+        }
+        catch (FileNotFoundException e)
+        {
+            System.out.println("File not found" + e);
+        }
+        catch (IOException ioe)
+        {
+            System.out.println("Exception while writing file " + ioe);
+        }
+        finally
+        {
+            // close the streams using close method
+            try
+            {
+                if (fos != null)
+                {
+                    fos.close();
+                }
+            }
+            catch (IOException ioe)
+            {
+                System.out.println("Error while closing stream: " + ioe);
+            }
+        }
+        return false;
+    }
+
+    static byte[] readBytesFromStream(InputStream in)
+    {
+        // this dynamically extends to take the bytes you read
+        ByteArrayOutputStream byteBuffer = new ByteArrayOutputStream();
+        // this is storage overwritten on each iteration with bytes
+        int bufferSize = 1024;
+        byte[] buffer = new byte[bufferSize];
+        // we need to know how may bytes were read to write them to the byteBuffer
+        int len;
+        try
+        {
+            while ((len = in.read(buffer)) != -1)
+            {
+                byteBuffer.write(buffer, 0, len);
+            }
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+        finally
+        {
+            try
+            {
+                byteBuffer.close();
+            }
+            catch (IOException e)
+            {
+                e.printStackTrace();
+            }
+            if (in != null)
+            {
+                try
+                {
+                    in.close();
+                }
+                catch (IOException e)
+                {
+                    e.printStackTrace();
+                }
+            }
+        }
+        // and then we can return your byte array.
+        return byteBuffer.toByteArray();
+    }
+
+
+    private synchronized void deleteKey(KeyStore keyStore, String keyAddr)
+    {
+        try
+        {
+            String encryptedBytesPath = getEncryptedFilePath(keyAddr);
```

Comment:
If deleting files causes an exception, the flow never reaches key deletion from keystore. Make all deletions not contingent on each other, i.e., delete as much as possible rather than "aborting" if a preceding delete operation fails.